crates/dbsp/src/circuit/metrics.rs

```diff
@@ -57,9 +66,290 @@ pub const COMPACTION_DURATION: &str = "file.compaction_duration";
 /// Time a worker was stalled waiting for more merges to complete.
 pub const COMPACTION_STALL_TIME: &str = "file.compaction_stall_time";
 
-/// Number of records dropped due to LATENES annotations
+/// Number of records dropped due to LATENESS annotations
 pub const TOTAL_LATE_RECORDS: &str = "records.late";
 
+/// Runtime in microseconds of an Operator evaluation
+pub const OPERATOR_EVAL_DURATION: &str = "operator.runtime_micros";
+
+/// Holds a map of the [`DBSPMetric`] and their equivalent [`metrics`] type.
+///
+/// [`RwLock`]s here should be very efficient, as reading is the most common operation.
+/// Writing is only necessary while adding a new metric.
+///
+/// Updating the inner type in the map doesn't require mutable access due to the interior
+/// mutability of the [`Counter`], [`Gauge`] and [`Histogram`] types.
+#[allow(unused)]
+#[derive(Default)]
+struct MetricsRecorder {
+    counters: RwLock<HashMap<DBSPMetric, Counter>>,
```

Comment:
The entire `DBSPMetric` should not be used as a key in the map. This data structure includes the `labels` array, which will make key comparison unnecessarily expensive, as it is not one of the attributes that uniquely identifies the the metric.